### PR TITLE
Deployment script module - Wait

### LINKED ISSUE #114: [Module Proposal]: Deployment delay
### Have you checked this module does not already exist in the registry?

Yes

### Why is the module needed?

When deploying some Azure services together that use AzureAD for RBAC, it is common to receive errors regarding propagation time.

> "Caller is not authorized to perform action on resource. If role assignments, deny assignments or role definitions were changed recently, please observe propagation time."

An example is when Azure Application Gateway is created that uses Key Vault for its SSL Certificates. On some tenants, success rate is at about 20%.

### Module path

deployment-scripts/wait

### Describe the module

This new module would create a sleep for a configurable number of seconds to be used in other resources `DependsOn` in order to delay resource creation. EG. [wait.bicep](https://github.com/Azure/azure-quickstart-templates/pull/12524/files#diff-cbe0f293f3aec177575c790878a3fb776fa65342455911c10f346ffef798217e)

### PR DESCRIPTION
**Pull request by Gordonby**
## Description

This new module would create a sleep for a configurable number of seconds to be used in other resources DependsOn in order to delay resource creation. 

Closes #114 

## Adding a new module

- [x] A proposal has been submitted and approved.
- [x] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

### CODE REVIEW

**Comment by Gordonby** [comment]
@shenglol - i've got an interesting CI error.  Could you have a look?

**Comment by shenglol** [comment]
@Gordonby Sure thing. The GitHub token has expired, and I updated it. The GitHub token was added when the repo was private so the CI can download files from pull requests. Since the repo is now public, I'll update the script to not use GitHub token so that we won't hit the same error again.

**Review by shenglol — APPROVED** [review]
:shipit:

**Comment by shenglol** [comment]
Need to onboard MCR manifest. Will merge it after that's done.

File: modules/deployment-scripts/wait/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Deployment Delay
+
+A Deployment Script that introduces a delay to the deployment process.
+
+## Description
+
+{{ Add detailed description for the module. }}
```
Comment (shenglol) [inline modules/deployment-scripts/wait/README.md:7]: @Gordonby sorry I missed this when I approve it. Can you add detailed description for the module? This is introduced in the latest version of the registry module tool.

**Review by shenglol — CHANGES_REQUESTED** [review]
Please see https://github.com/Azure/bicep-registry-modules/pull/115/files#r924820892.

File: modules/deployment-scripts/wait/README.md (hunk 1 above)
Comment (Gordonby) [reply modules/deployment-scripts/wait/README.md]: @shenglol  - done.

**Review by shenglol — APPROVED** [review]

**Pull request merged**